integreat_cms/static/src/js/pages/page-api-token.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Generate and (un)set the API token.
+ */
+
+// Set event handler
+window.addEventListener("load", () => {
+    document.getElementById("checkbox_api_token")?.addEventListener("change", setApiToken);
+    let checkbox = document.getElementById("checkbox_api_token") as HTMLInputElement;
+    let text = document.getElementById("id_api_token") as HTMLInputElement;
+    if (text.value == ""){
+      checkbox.checked = false;
+    } else {
+      checkbox.checked = true;
+    }
+});
+
+/**
+ * Generate 24 characters API token and insert it into the text field
+ *
+ * @param {target} eventTarget - The changed file input
+ */
+function setApiToken({ target }: Event) {
+    let checkBox = document.getElementById("checkbox_api_token") as HTMLInputElement;
+    let text = document.getElementById("id_api_token") as HTMLInputElement;
+    let random = "";
+    if (checkBox.checked == true){
+      let random = "";
+      const characters ='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+      for ( let i = 0; i < 24; i++ ) {
+        random += characters.charAt(Math.floor(Math.random() * characters.length));
+      }
+      text.value = random;
```

Comment:
```suggestion
    if (checkBox.checked){
      text.value = window.crypto.randomUUID();
      apiTokenContainer.classList.remove("hidden");
```
Since `Math.random()` is not cryptocraphically secure, I suggest using `window.crypto.randomUUID()` for this task (https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID). This only has the side-effect that the tokens are now 36 instead of 24 chars (but I see no reason why we should lower the entropy here). And it requires an update of  typescript (run `npm update`) because the type definition of this module was recently updated.
Oh, and [browser support](https://caniuse.com/mdn-api_crypto_randomuuid) isn't optimal, but I'm not sure if webpack does any magic here to make it available in older browsers... we should probably test this and maybe provide a fallback?

Reply:
I fit's too hard to do it cryptographically secure in the browser, we could also just say that the user has to enable the checkbox and submit the form, then the server calculates a secret, and the user can copy it afterwards.

Reply:
Yes, using a UUID is a good idea.